### DISCUSSION TITLE
OnRecordBeforeUpdateRequest behavior, how to transactionally insert two records in different collections and return success

### DISCUSSION THREAD

**Discussion by Irvingouj** [pocketbase/pocketbase#5272] [Q&A] [answered]
Here is my code
```go
app.OnRecordBeforeUpdateRequest("orders").Add(func(e *core.RecordUpdateEvent) error {
		if (e.Record.OriginalCopy().Get("status") != "created") || (e.Record.Get("status") != "paid") {
			return apis.NewBadRequestError("Invalid status change", nil)
		}

		currentUser, err := utils.GetCurrentLoginRecord(e.HttpContext, app)
		if err != nil {
			return err
		}

		err := app.Dao().RunInTransaction(func(tx *daos.Dao) error {
			user_point_events_collection := tx.FindCollectionByNameOrId("user_points_event")
			// insert user_points_event here...
			// update the order status here...
			return nil
		})

		return err
	})
```

my questions is that, I understand by returning an error, if prevents the record form being created. when returns nil, it continues with the recording creation.

how do not return an error but still prevent it from updating the new record?

**Comment by ganigeorgiev** [comment]
I'm not sure I follow. The above code sample is confusing and it is not clear what you are asking or trying to do.

If you are trying to wrap the currently updated order from the request as part of a transaction, then it is not possible at the moment, aka. you can't have _"update the order status here..."_ inside your transaction (this is supported with the refactoring).

**Reply by Irvingouj** [reply]
To make it clear.
Is it possible to intercept the update of orders table without returning an error in the event handler?

**Reply by Irvingouj** [reply]
I guess the answer is no at the moment. So the better way to do it is to create a custom endpoint, or do you think there might be another way around?

**Reply by ganigeorgiev** [reply, marked as answer]
> To make it clear. Is it possible to intercept the update of orders table without returning an error in the event handler?

This doesn't really make it more clear but I will not pursue the matter further.

I doubt that this is what you are asking but to answer directly your question - Yes, but at the moment the request operation cannot be wrapped in a transaction, ex.:

```go
app.OnRecordBeforeUpdateRequest("orders").Add(func(e *core.RecordUpdateEvent) error {
	e.Record.Set("someField", "example")

	return nil
})
```

**Reply by Irvingouj** [reply]
Thank you
